### ISSUE TITLE
Build fails for nanoatmega328

### ISSUE THREAD

**Issue by AlmightyFrog** [Schallbert/Tonuino_UseEncoder#2]
Hi,

i currently have only a windows system available, so not sure if it is causing an issue.
Following output is from simple build of nanoatmega328 env:

```
> Executing task: C:\Users\User\.platformio\penv\Scripts\platformio.exe run --environment nanoatmega328 <

Processing nanoatmega328 (platform: atmelavr; board: nanoatmega328; framework: arduino)
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/atmelavr/nanoatmega328.html
PLATFORM: Atmel AVR (3.4.0) > Arduino Nano ATmega328
HARDWARE: ATMEGA328P 16MHz, 2KB RAM, 30KB Flash
DEBUG: Current (avr-stub) On-board (avr-stub, simavr)
PACKAGES: 
 - framework-arduino-avr 5.1.0
 - toolchain-atmelavr 1.70300.191015 (7.3.0)
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 23 compatible libraries
Scanning dependencies...
Dependency Graph
|-- <TimerOne> 1.1.0
|-- <SoftwareSerial> 1.0.0
|-- <ClickEncoder> 1.1.0
|-- <DFPlayer Mini Mp3 by Makuna> 1.1.0
|-- <MFRC522> 1.4.10
|   |-- <SPI> 1.0
|-- <Loader>
|   |-- <Arduino>
|   |   |-- <Arduino_HardwareAbstraction>
|   |   |-- <EEPROM> 2.0
|   |   |-- <Config>
|   |   |-- <ClickEncoder> 1.1.0
|   |   |-- <MessageHandler>
|   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |-- <Mp3>
|   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <Config>
|   |   |   |-- <Nfc>
|   |   |   |   |-- <MessageHandler>
|   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <Config>
|   |   |   |   |-- <MFRC522> 1.4.10
|   |   |   |   |   |-- <SPI> 1.0
|   |   |   |   |-- <Folder>
|   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |   |-- <MessageHandler>
|   |   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |   |-- <Config>
|   |   |   |-- <PowerManager>
|   |   |   |   |-- <Config>
|   |   |   |   |-- <Utilities>
|   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <MessageHandler>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <Folder>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <MessageHandler>
|   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <Config>
|   |   |   |-- <Utilities>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |-- <UserInput>
|   |   |   |-- <MessageHandler>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |-- <DFPlayer Mini Mp3 by Makuna> 1.1.0
|   |   |-- <SoftwareSerial> 1.0.0
|   |-- <MessageHandler>
|   |   |-- <Arduino_HardwareAbstraction>
|   |-- <Mp3>
|   |   |-- <Arduino_HardwareAbstraction>
|   |   |-- <Config>
|   |   |-- <Nfc>
|   |   |   |-- <MessageHandler>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <Config>
|   |   |   |-- <MFRC522> 1.4.10
|   |   |   |   |-- <SPI> 1.0
|   |   |   |-- <Folder>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <MessageHandler>
|   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <Config>
|   |   |-- <PowerManager>
|   |   |   |-- <Config>
|   |   |   |-- <Utilities>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |-- <MessageHandler>
|   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |-- <Folder>
|   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <MessageHandler>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <Config>
|   |   |-- <Utilities>
|   |   |   |-- <Arduino_HardwareAbstraction>
|   |-- <Nfc>
|   |   |-- <MessageHandler>
|   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |-- <Arduino_HardwareAbstraction>
|   |   |-- <Config>
|   |   |-- <MFRC522> 1.4.10
|   |   |   |-- <SPI> 1.0
|   |   |-- <Folder>
|   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <MessageHandler>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <Config>
|   |-- <PowerManager>
|   |   |-- <Config>
|   |   |-- <Utilities>
|   |   |   |-- <Arduino_HardwareAbstraction>
|   |-- <Utilities>
|   |   |-- <Arduino_HardwareAbstraction>
|   |-- <VoiceMenu>
|   |   |-- <Arduino_HardwareAbstraction>
|   |   |-- <Config>
|   |   |-- <Mp3>
|   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <Config>
|   |   |   |-- <Nfc>
|   |   |   |   |-- <MessageHandler>
|   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <Config>
|   |   |   |   |-- <MFRC522> 1.4.10
|   |   |   |   |   |-- <SPI> 1.0
|   |   |   |   |-- <Folder>
|   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |   |-- <MessageHandler>
|   |   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |   |-- <Config>
|   |   |   |-- <PowerManager>
|   |   |   |   |-- <Config>
|   |   |   |   |-- <Utilities>
|   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <MessageHandler>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <Folder>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <MessageHandler>
|   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <Config>
|   |   |   |-- <Utilities>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |-- <Folder>
|   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <MessageHandler>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <Config>
|   |   |-- <Nfc>
|   |   |   |-- <MessageHandler>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <Config>
|   |   |   |-- <MFRC522> 1.4.10
|   |   |   |   |-- <SPI> 1.0
|   |   |   |-- <Folder>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <MessageHandler>
|   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <Config>
|   |   |-- <PowerManager>
|   |   |   |-- <Config>
|   |   |   |-- <Utilities>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |-- <MessageHandler>
|   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |-- <Utilities>
|   |   |   |-- <Arduino_HardwareAbstraction>
|   |-- <Tonuino>
|   |   |-- <Mp3>
|   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <Config>
|   |   |   |-- <Nfc>
|   |   |   |   |-- <MessageHandler>
|   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <Config>
|   |   |   |   |-- <MFRC522> 1.4.10
|   |   |   |   |   |-- <SPI> 1.0
|   |   |   |   |-- <Folder>
|   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |   |-- <MessageHandler>
|   |   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |   |-- <Config>
|   |   |   |-- <PowerManager>
|   |   |   |   |-- <Config>
|   |   |   |   |-- <Utilities>
|   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <MessageHandler>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <Folder>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <MessageHandler>
|   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <Config>
|   |   |   |-- <Utilities>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |-- <Nfc>
|   |   |   |-- <MessageHandler>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <Config>
|   |   |   |-- <MFRC522> 1.4.10
|   |   |   |   |-- <SPI> 1.0
|   |   |   |-- <Folder>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <MessageHandler>
|   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <Config>
|   |   |-- <UserInput>
|   |   |   |-- <MessageHandler>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |-- <VoiceMenu>
|   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <Config>
|   |   |   |-- <Mp3>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <Config>
|   |   |   |   |-- <Nfc>
|   |   |   |   |   |-- <MessageHandler>
|   |   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |   |-- <Config>
|   |   |   |   |   |-- <MFRC522> 1.4.10
|   |   |   |   |   |   |-- <SPI> 1.0
|   |   |   |   |   |-- <Folder>
|   |   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |   |   |-- <MessageHandler>
|   |   |   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |   |   |-- <Config>
|   |   |   |   |-- <PowerManager>
|   |   |   |   |   |-- <Config>
|   |   |   |   |   |-- <Utilities>
|   |   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <MessageHandler>
|   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <Folder>
|   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |   |-- <MessageHandler>
|   |   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |   |-- <Config>
|   |   |   |   |-- <Utilities>
|   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <Folder>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <MessageHandler>
|   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <Config>
|   |   |   |-- <Nfc>
|   |   |   |   |-- <MessageHandler>
|   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |-- <Config>
|   |   |   |   |-- <MFRC522> 1.4.10
|   |   |   |   |   |-- <SPI> 1.0
|   |   |   |   |-- <Folder>
|   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |   |-- <MessageHandler>
|   |   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |   |   |-- <Config>
|   |   |   |-- <PowerManager>
|   |   |   |   |-- <Config>
|   |   |   |   |-- <Utilities>
|   |   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <MessageHandler>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <Utilities>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |-- <Folder>
|   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <MessageHandler>
|   |   |   |   |-- <Arduino_HardwareAbstraction>
|   |   |   |-- <Config>
|   |   |-- <MessageHandler>
|   |   |   |-- <Arduino_HardwareAbstraction>
|-- <Config>
Building in release mode
Compiling .pio\build\nanoatmega328\src\main.cpp.o
Compiling .pio\build\nanoatmega328\lib5fa\TimerOne\TimerOne.cpp.o
Compiling .pio\build\nanoatmega328\lib463\SoftwareSerial\SoftwareSerial.cpp.o
Compiling .pio\build\nanoatmega328\lib282\ClickEncoder\ClickEncoder.cpp.o
Compiling .pio\build\nanoatmega328\lib1fb\SPI\SPI.cpp.o
Compiling .pio\build\nanoatmega328\lib349\MFRC522\MFRC522.cpp.o
Compiling .pio\build\nanoatmega328\lib349\MFRC522\MFRC522Extended.cpp.o
Compiling .pio\build\nanoatmega328\lib447\MessageHandler\MessageHandler_implementation.cpp.o
In file included from lib\Loader/../Arduino/DFMiniMp3_implementation.h:14:0,
                 from lib\Loader/Loader.h:12,
                 from src\main.cpp:5:
.pio\libdeps\nanoatmega328\DFPlayer Mini Mp3 by Makuna\src/DFMiniMp3.h: In instantiation of 'uint16_t DFMiniMp3<T_SERIAL_METHOD, T_NOTIFICATION_METHOD>::listenForReply(uint8_t) [with T_SERIAL_METHOD = SoftwareSerial; T_NOTIFICATION_METHOD = Mp3Notify; uint16_t = unsigned int; uint8_t = unsigned char]':
.pio\libdeps\nanoatmega328\DFPlayer Mini Mp3 by Makuna\src/DFMiniMp3.h:105:27:   required from 'void DFMiniMp3<T_SERIAL_METHOD, T_NOTIFICATION_METHOD>::loop() [with T_SERIAL_METHOD = SoftwareSerial; T_NOTIFICATION_METHOD = Mp3Notify]'
lib\Loader/../Arduino/DFMiniMp3_implementation.h:87:26:   required from here
.pio\libdeps\nanoatmega328\DFPlayer Mini Mp3 by Makuna\src/DFMiniMp3.h:511:62: error: no matching function for call to 'Mp3Notify::OnPlayFinished(DFMiniMp3<SoftwareSerial, Mp3Notify>&, DfMp3_PlaySources, uint16_t&)'
                         T_NOTIFICATION_METHOD::OnPlayFinished(*this, DfMp3_PlaySources_Usb, replyArg);
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from lib\Loader/Loader.h:12:0,
                 from src\main.cpp:5:
Archiving .pio\build\nanoatmega328\lib5fa\libTimerOne.a
lib\Loader/../Arduino/DFMiniMp3_implementation.h:29:17: note: candidate: static void Mp3Notify::OnPlayFinished(DfMp3_PlaySources, uint16_t)
Archiving .pio\build\nanoatmega328\lib463\libSoftwareSerial.a
     static void OnPlayFinished(DfMp3_PlaySources src, uint16_t track)
                 ^~~~Archiving .pio\build\nanoatmega328\lib1fb\libSPI.a
Archiving .pio\build\nanoatmega328\lib447\libMessageHandler.a
~~~~~~~~~~
lib\Loader/../Arduino/DFMiniMp3_implementation.h:29:17: note:   candidate expects 2 arguments, 3 provided
In file included from lib\Loader/../Arduino/DFMiniMp3_implementation.h:14:0,
                 from lib\Loader/Loader.h:12,
                 from src\main.cpp:5:
.pio\libdeps\nanoatmega328\DFPlayer Mini Mp3 by Makuna\src/DFMiniMp3.h:515:62: error: no matching function for call to 'Mp3Notify::OnPlayFinished(DFMiniMp3<SoftwareSerial, Mp3Notify>&, DfMp3_PlaySources, uint16_t&)'
                         T_NOTIFICATION_METHOD::OnPlayFinished(*this, DfMp3_PlaySources_Sd, replyArg);
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from lib\Loader/Loader.h:12:0,
                 from src\main.cpp:5:
lib\Loader/../Arduino/DFMiniMp3_implementation.h:29:17: note: candidate: static void Mp3Notify::OnPlayFinished(DfMp3_PlaySources, uint16_t)
     static void OnPlayFinished(DfMp3_PlaySources src, uint16_t track)
                 ^~~~~~~~~~~~~~
lib\Loader/../Arduino/DFMiniMp3_implementation.h:29:17: note:   candidate expects 2 arguments, 3 provided
In file included from lib\Loader/../Arduino/DFMiniMp3_implementation.h:14:0,
                 from lib\Loader/Loader.h:12,
                 from src\main.cpp:5:
.pio\libdeps\nanoatmega328\DFPlayer Mini Mp3 by Makuna\src/DFMiniMp3.h:519:62: error: no matching function for call to 'Mp3Notify::OnPlayFinished(DFMiniMp3<SoftwareSerial, Mp3Notify>&, DfMp3_PlaySources, uint16_t&)'
                         T_NOTIFICATION_METHOD::OnPlayFinished(*this, DfMp3_PlaySources_Flash, replyArg);
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from lib\Loader/Loader.h:12:0,
                 from src\main.cpp:5:
lib\Loader/../Arduino/DFMiniMp3_implementation.h:29:17: note: candidate: static void Mp3Notify::OnPlayFinished(DfMp3_PlaySources, uint16_t)
     static void OnPlayFinished(DfMp3_PlaySources src, uint16_t track)
                 ^~~~~~~~~~~~~~
lib\Loader/../Arduino/DFMiniMp3_implementation.h:29:17: note:   candidate expects 2 arguments, 3 provided
In file included from lib\Loader/../Arduino/DFMiniMp3_implementation.h:14:0,
                 from lib\Loader/Loader.h:12,
                 from src\main.cpp:5:
.pio\libdeps\nanoatmega328\DFPlayer Mini Mp3 by Makuna\src/DFMiniMp3.h:524:66: error: no matching function for call to 'Mp3Notify::OnPlaySourceOnline(DFMiniMp3<SoftwareSerial, Mp3Notify>&, DfMp3_PlaySources)'
                         T_NOTIFICATION_METHOD::OnPlaySourceOnline(*this, static_cast<DfMp3_PlaySources>(replyArg));
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from lib\Loader/Loader.h:12:0,
                 from src\main.cpp:5:
lib\Loader/../Arduino/DFMiniMp3_implementation.h:34:17: note: candidate: static void Mp3Notify::OnPlaySourceOnline(uint16_t)
     static void OnPlaySourceOnline(uint16_t code)
                 ^~~~~~~~~~~~~~~~~~
lib\Loader/../Arduino/DFMiniMp3_implementation.h:34:17: note:   candidate expects 1 argument, 2 provided
In file included from lib\Loader/../Arduino/DFMiniMp3_implementation.h:14:0,
                 from lib\Loader/Loader.h:12,
                 from src\main.cpp:5:
.pio\libdeps\nanoatmega328\DFPlayer Mini Mp3 by Makuna\src/DFMiniMp3.h:529:68: error: no matching function for call to 'Mp3Notify::OnPlaySourceInserted(DFMiniMp3<SoftwareSerial, Mp3Notify>&, DfMp3_PlaySources)'
                         T_NOTIFICATION_METHOD::OnPlaySourceInserted(*this, static_cast<DfMp3_PlaySources>(replyArg));
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from lib\Loader/Loader.h:12:0,
                 from src\main.cpp:5:
lib\Loader/../Arduino/DFMiniMp3_implementation.h:39:17: note: candidate: static void Mp3Notify::OnPlaySourceInserted(uint16_t)
     static void OnPlaySourceInserted(uint16_t code)
                 ^~~~~~~~~~~~~~~~~~~~
lib\Loader/../Arduino/DFMiniMp3_implementation.h:39:17: note:   candidate expects 1 argument, 2 provided
In file included from lib\Loader/../Arduino/DFMiniMp3_implementation.h:14:0,
                 from lib\Loader/Loader.h:12,
                 from src\main.cpp:5:
.pio\libdeps\nanoatmega328\DFPlayer Mini Mp3 by Makuna\src/DFMiniMp3.h:534:67: error: no matching function for call to 'Mp3Notify::OnPlaySourceRemoved(DFMiniMp3<SoftwareSerial, Mp3Notify>&, DfMp3_PlaySources)'
                         T_NOTIFICATION_METHOD::OnPlaySourceRemoved(*this, static_cast<DfMp3_PlaySources>(replyArg));
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from lib\Loader/Loader.h:12:0,
                 from src\main.cpp:5:
lib\Loader/../Arduino/DFMiniMp3_implementation.h:44:17: note: candidate: static void Mp3Notify::OnPlaySourceRemoved(uint16_t)
     static void OnPlaySourceRemoved(uint16_t code)
                 ^~~~~~~~~~~~~~~~~~~
lib\Loader/../Arduino/DFMiniMp3_implementation.h:44:17: note:   candidate expects 1 argument, 2 provided
In file included from lib\Loader/../Arduino/DFMiniMp3_implementation.h:14:0,
                 from lib\Loader/Loader.h:12,
                 from src\main.cpp:5:
.pio\libdeps\nanoatmega328\DFPlayer Mini Mp3 by Makuna\src/DFMiniMp3.h:538:55: error: no matching function for call to 'Mp3Notify::OnError(DFMiniMp3<SoftwareSerial, Mp3Notify>&, uint16_t&)'
                         T_NOTIFICATION_METHOD::OnError(*this, replyArg);
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
In file included from lib\Loader/Loader.h:12:0,
                 from src\main.cpp:5:
lib\Loader/../Arduino/DFMiniMp3_implementation.h:24:17: note: candidate: static void Mp3Notify::OnError(uint16_t)
     static void OnError(uint16_t errorCode)
                 ^~~~~~~
lib\Loader/../Arduino/DFMiniMp3_implementation.h:24:17: note:   candidate expects 1 argument, 2 provided
In file included from lib\Loader/../Arduino/DFMiniMp3_implementation.h:14:0,
                 from lib\Loader/Loader.h:12,
                 from src\main.cpp:5:
.pio\libdeps\nanoatmega328\DFPlayer Mini Mp3 by Makuna\src/DFMiniMp3.h:552:51: error: no matching function for call to 'Mp3Notify::OnError(DFMiniMp3<SoftwareSerial, Mp3Notify>&, uint16_t&)'
                     T_NOTIFICATION_METHOD::OnError(*this, replyArg);
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~
In file included from lib\Loader/Loader.h:12:0,
                 from src\main.cpp:5:
lib\Loader/../Arduino/DFMiniMp3_implementation.h:24:17: note: candidate: static void Mp3Notify::OnError(uint16_t)
     static void OnError(uint16_t errorCode)
                 ^~~~~~~
lib\Loader/../Arduino/DFMiniMp3_implementation.h:24:17: note:   candidate expects 1 argument, 2 provided
*** [.pio\build\nanoatmega328\src\main.cpp.o] Error 1
============================================================================================================== [FAILED] Took 2.97 seconds ==============================================================================================================

Environment    Status    Duration
-------------  --------  ------------
nanoatmega328  FAILED    00:00:02.971
========================================================================================================= 1 failed, 0 succeeded in 00:00:02.971 =========================================================================================================The terminal process "C:\Users\User\.platformio\penv\Scripts\platformio.exe 'run', '--environment', 'nanoatmega328'" terminated with exit code: 1.
```

Any Idea what I am doing wrong, or did maybe change a lib or so?
Btw, it is a freshly installed Platform IO instance.

Thanks!

**Comment by Schallbert** [comment]
Hey @AlmightyFrog,

Your assumption is correct: the API of the DFplayer mini library's callback methods must have changed.  Normally this shouldn't be an issue as I'm explicitly mentioning the lib version, but it looks like the lib has been re-released under the same tag with different content. 

Unfortunately, I'll be busy all of next week so that I cannot offer a quick fix adapting my solution to the new api. The easiest option for you would be to try with an earlier lib version, around end of 2020 release or so.
I'll let you know once I adapt my code to the new lib version.

Cheers,  Schallbert

**Comment by Schallbert** [comment]
Quick fix: in your `platformio.ini`, remove the `^` sign to indicate that you don't want the library dependency finder to auto-update to the latest available lib version if 1.0.7 is not already installed. Like this:
`makuna/DFPlayer Mini Mp3 by Makuna @ 1.0.7  # Mp3 Player`
That way, compiling shoud succeed.
![image](https://user-images.githubusercontent.com/48652779/141526873-ffdd192a-ec6a-4a30-918b-75a5483ba45a.png)

Cheers, Schallbert

**Comment by Schallbert** [comment]
Permanently fixed with https://github.com/Schallbert/Tonuino_alternativeFirmware/commit/9591b02d6b506c37e0e43561846216120baa3bbd . Updated to latest lib API. Closing issue.
